src/main/java/intellistart/interviewplanning/controllers/JwtAuthenticationController.kt

```diff
@@ -0,0 +1,124 @@
+package intellistart.interviewplanning.controllers
+
+import intellistart.interviewplanning.controllers.dto.CandidateDto
+import intellistart.interviewplanning.controllers.dto.FacebookOauthInfoDto
+import intellistart.interviewplanning.controllers.dto.JwtRequest
+import intellistart.interviewplanning.controllers.dto.JwtResponse
+import intellistart.interviewplanning.exceptions.SecurityException
+import intellistart.interviewplanning.exceptions.SecurityException.SecurityExceptionProfile
+import intellistart.interviewplanning.model.user.UserService
+import intellistart.interviewplanning.security.JwtUserDetails
+import intellistart.interviewplanning.security.JwtUserDetailsService
+import intellistart.interviewplanning.utils.FacebookUtil
+import intellistart.interviewplanning.utils.FacebookUtil.FacebookScopes
+import intellistart.interviewplanning.utils.JwtUtil
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.ResponseEntity
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.BadCredentialsException
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.Authentication
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.bind.annotation.CrossOrigin
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.client.RestClientException
+import redis.clients.jedis.JedisPooled
+
+/**
+ * Controller for authentication and authenticated requests.
+ */
+@RestController
+@CrossOrigin
+class JwtAuthenticationController(
+    private val authenticationManager: AuthenticationManager,
+    private val jwtUtil: JwtUtil,
+    private val userDetailsService: JwtUserDetailsService,
+    private val facebookUtil: FacebookUtil,
+    private val userService: UserService,
+    private val jedis: JedisPooled
+) {
+    @Value("\${jwt.caching}")
+    private val jwtValidity: Long = 0
+
+    private val logger: Logger = LogManager.getLogger(JwtAuthenticationController::class.java)
```

Comment:
```suggestion
    companion object {
        private val logger: Logger = LogManager.getLogger(JwtAuthenticationController::class.java)
    }
```

Note: the companion object must be declared at the end of class